app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt

```diff
@@ -0,0 +1,177 @@
+package org.p2p.wallet.send.ui.search
+
+import android.os.Bundle
+import android.view.View
+import androidx.activity.addCallback
+import androidx.appcompat.widget.SearchView
+import androidx.core.view.isInvisible
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.LinearLayoutManager
+import org.koin.android.ext.android.get
+import org.koin.android.ext.android.inject
+import org.koin.core.parameter.parametersOf
+import org.p2p.uikit.utils.attachAdapter
+import org.p2p.wallet.R
+import org.p2p.wallet.common.analytics.constants.ScreenNames
+import org.p2p.wallet.common.analytics.interactor.ScreensAnalyticsInteractor
+import org.p2p.wallet.common.mvp.BaseMvpFragment
+import org.p2p.wallet.databinding.FragmentNewSearchBinding
+import org.p2p.wallet.qr.ui.ScanQrFragment
+import org.p2p.wallet.send.model.SearchResult
+import org.p2p.wallet.send.ui.main.SendFragment
+import org.p2p.wallet.send.ui.search.adapter.SearchAdapter
+import org.p2p.wallet.utils.addFragment
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.getClipboardText
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.unsafeLazy
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+
+class NewSearchFragment :
+    BaseMvpFragment<NewSearchContract.View, NewSearchContract.Presenter>(R.layout.fragment_new_search),
+    NewSearchContract.View {
+
+    companion object {
+        private const val EXTRA_USERNAMES = "EXTRA_USERNAMES"
+
+        fun create(usernames: List<SearchResult>? = null): NewSearchFragment =
```

Comment:
always null, no?

Reply:
if it's for future 'popular recipients' there is not need for this arg either. 
you can fetch those from the repository

Reply:
it's when you go from send fragment and you already have some selected recipients

if this won't be needed we will remove it